compiler/backend/jsgen.nim

```diff
@@ -848,10 +848,8 @@ proc genTry(p: PProc, n: PNode, r: var TCompRes) =
           # If this is a ``except exc as sym`` branch there must be no following
           # nodes
           doAssert orExpr == nil
-        elif it.kind == nkType:
-          throwObj = it
-        else:
-          internalError(p.config, n.info, "genTryStmt")
+        p.config.internalAssert(it.kind == nkType, n.info, "genTryStmt")
+        throwObj = it
```

Comment:
Doesn't this change the logic completely?

Reply:
No, not at all.